x/incentives/keeper/distribute.go

```diff
@@ -384,59 +408,20 @@ func (k Keeper) Distribute(ctx sdk.Context, gauges []types.Gauge) (sdk.Coins, er
 	locksByDenomCache := make(map[string][]lockuptypes.PeriodLock)
 	totalDistributedCoins := sdk.NewCoins()
 
-	// get pool Id corresponding to the gaugeId
-	incentiveParams := k.GetParams(ctx).DistrEpochIdentifier
-	currentEpoch := k.ek.GetEpochInfo(ctx, incentiveParams)
-
 	for _, gauge := range gauges {
 		var gaugeDistributedCoins sdk.Coins
-		pool, err := k.GetPoolFromGaugeId(ctx, gauge.Id, currentEpoch.Duration)
-		// Note: getting NoPoolAssociatedWithGaugeError implies that there is no pool associated with the gauge but we still want to distribute.
-		// This happens with superfluid gauges which are not connected to any specific pool directly but, instead,
-		// via an intermediary account.
-		if err != nil && !errors.Is(err, poolincentivestypes.NoPoolAssociatedWithGaugeError{GaugeId: gauge.Id, Duration: currentEpoch.Duration}) {
-			// TODO: add test case to cover this
-			return nil, err
-		} else if pool != nil && pool.GetType() == poolmanagertypes.Concentrated {
-			// only want to run this logic if the gaugeId is associated with CL PoolId
-			for _, coin := range gauge.Coins {
-				// emissionRate calculates amount of tokens to emit per second
-				// for ex: 10000tokens to be distributed over 1day epoch will be 1000 tokens ÷ 86,400 seconds ≈ 0.01157 tokens per second (truncated)
-				// Note: reason why we do millisecond conversion is because floats are non-deterministic so if someone refactors this and accidentally
-				// uses the return of currEpoch.Duration.Seconds() in math operations, this will lead to an app hash.
-				emissionRate := sdk.NewDecFromInt(coin.Amount).QuoTruncate(sdk.NewDec(currentEpoch.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
-				err := k.distributeConcentratedLiquidity(ctx,
-					pool.GetId(),
-					k.ak.GetModuleAddress(types.ModuleName),
-					coin,
-					emissionRate,
-					gauge.GetStartTime(),
-					// Note that the minimum uptime does not affect the distribution of incentives from the gauge and
-					// thus can be any value authorized by the CL module.
-					types.DefaultConcentratedUptime,
-					gauge,
-				)
-				if err != nil {
-					return nil, err
-				}
-				gaugeDistributedCoins = gaugeDistributedCoins.Add(coin)
-			}
```

Comment:
Note:
all previously added CL logic was reverted and instead moved to `distributeInternal`. This was done so that we can share the logic for handling external gauges. Previous logic only worked for internal gauges paid over 1 epoch.

By moving to `distributeInternal`, we can share a lot of the existing boilerplating for handing both perpetual and non-perpetual gauges